### PR TITLE
Loki: Fix error handling

### PR DESCRIPTION
**Pull request by torkelo**
Noticed that loki error handling is broken in 7.0 & 7.1 (and master), unrelated to backend_srv observable rewrite. 

* Removed the error handling in loki data source, seems to be old and not relevant anymore as backendSrv has done that data string => obj translation for a long time 
* There is still some logic of capturing 404 error in some cases and not letting this be seen as an error, so kept that logic
* Removed cancelled filtering as the backendSrv rewrite now makes cancelled requests result in errors so no need to filter these out (So this PR cannot be cherry picked to 7.1) 

Before: (7.0.6)
Screenshot from 2020-07-13 15-15-36

After:
Screenshot from 2020-07-13 15-14-33

This problem could exist for Prometheus instead.

There are some strange safeStringifyValue that I do not really understand the origin for. 

To fix this for 7.1 we need a slightly modified PR that does not rely on the backendSrv changes in master/7.2   (specifically the changes to cancellation that are now seen as errors)

### CODE REVIEW

File: public/app/plugins/datasource/loki/datasource.ts [hunk 1]
```diff
@@ -498,42 +495,13 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
...
-  processError = (err: any, target: LokiQuery): DataQueryError => {
-    const error: DataQueryError = {
-      message: (err && err.statusText) || 'Unknown error during query transaction. Please check JS console logs.',
-      refId: target.refId,
-    };
-
-    if (err.data) {
-      if (typeof err.data === 'string') {
-        if (err.data.includes('escape') && target.expr.includes('\\')) {
-          error.message = `Error: ${err.data}. Make sure that all special characters are escaped with \\. For more information on escaping of special characters visit LogQL documentation at https://github.com/grafana/loki/blob/master/docs/logql.md.`;
```
Comment (aocenas) [inline public/app/plugins/datasource/loki/datasource.ts:519]: We actually added this message 4 days ago https://github.com/grafana/grafana/pull/26136. Not sure if it makes sense to remove all of this code. cc @ivanahuckova.

Comment (torkelo) [reply public/app/plugins/datasource/loki/datasource.ts:519]: err.data is never a string here, backendSrv transforms it (same was true before my backendSrv refactor as far as I can tell) 

https://github.com/grafana/grafana/blob/master/public/app/core/services/backend_srv.ts#L248

Comment (torkelo) [reply public/app/plugins/datasource/loki/datasource.ts:519]: but please add this back if that is the case, but check err.data.message  instead

Comment (aocenas) [reply public/app/plugins/datasource/loki/datasource.ts:519]: @ivanahuckova can you check this change and see if we should add that message somewhere else?

Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/datasource.ts:519]: yeah, I will look at it the soonest I can (after Elastic PR).

Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/datasource.ts:519]: I hope it is okay, I have added a commit to this PR that keeps the custom escape error processing https://github.com/grafana/grafana/pull/26291/commits/b428052e423f84321f5ccd3b1dfb44bc0949582e. If not, I can still revert and create separate PR.

Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/datasource.ts:519]: I have also fixed the failing test 466b01b.

**Review by ivanahuckova — APPROVED** [review]
I have tested this and LGTM. Everything works as expected. I hope it is okay that I have added 2 commits fixing the test and adding custom error processing.

**Comment by torkelo** [comment]
Nice, thanks for the help getting this mergable!

**Pull request merged**